solidity/contracts/bridge/Bridge.sol

```diff
@@ -478,13 +495,43 @@ contract Bridge is Ownable, EcdsaWalletOwner {
         wallets.registerNewWallet(ecdsaWalletID, publicKeyX, publicKeyY);
     }
 
-    // TODO: Documentation.
+    /// @notice A callback function that is called by the ECDSA Wallet Registry
+    ///         once a wallet heartbeat failure is detected.
+    /// @param ecdsaWalletID Wallet's unique identifier.
+    /// @param publicKeyX Wallet's public key's X coordinate.
+    /// @param publicKeyY Wallet's public key's Y coordinate.
+    /// @dev Requirements:
+    ///      - The only caller authorized to call this function is `registry`
+    ///      - Wallet must be in Live state
     function __ecdsaWalletHeartbeatFailedCallback(
+        /* solhint-disable-next-line no-unused-vars */
         bytes32 ecdsaWalletID,
         bytes32 publicKeyX,
         bytes32 publicKeyY
     ) external override {
-        // TODO: Implementation.
+        wallets.notifyWalletHeartbeatFailed(publicKeyX, publicKeyY);
+    }
+
+    /// @notice Notifies about wallet exhaustion and triggers the
+    ///         wallet closure process.
```

Comment:
Let's be more specific. My proposition: 

Notifies that the wallet is either old enough or has too few satoshis left and qualifies to be closed.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/176/commits/c10e1a1b8f474a801681ec1766337e624364d961